src/adapters/kafka/index.ts

```diff
@@ -42,18 +40,29 @@ class KafkaAdapter extends Adapter {
     await consumer.subscribe({ topics: subscribedChannels, fromBeginning: true })
     await consumer.run({
       eachMessage: async ({ topic, partition, message }) => {
-        const msg = this._createMessage(topic, message)
+        const msg = this._createMessage(topic, partition, message)
         this.emit('message', msg, consumer)
       },
-    }) 
+    })
+    
+    const producer = kafka.producer()
+    await producer.connect()
+    await producer.send({
+      topic: 'glee-topic',
+      messages: [{ value: 'Hello, Glee!' }],
+    })
```

Comment:
If you put the producer code here, it will send a "Hello, Glee" message to Kafka every time we connect to the broker. And that's not what we want, right? We want to send a message (GleeMessage) to Kafka every time someone calls the "send" function below. So you have to move your code over there.

Reply:
yess i actually put the producer part in the send function only but then the kafka client is declared over here and that was not accessible in the send function below, but i guess i have to declare that globally so that its accessible to each function